### DISCUSSION TITLE
Ability to add category folders to the customer POS app

### DISCUSSION THREAD

**Discussion by Zaxounette** [btcpayserver/btcpayserver#4468] [Ideas / Feature Requests]
I just got a request through twitter DM on this. Tried a quick issue check, didn't find any resembling this.

The request is to be able to have folder of sorts in the POS app itself.
The user requesting this is in the food business.

So for example

* Burgers (folder)
  * Item 1
  * Item 2
* Drinks (folder)
  * Item 1
  * Item 2
* Etc ...

Their goal is to have the customer be able to order themselves on a POS tablet, with a ticket being printed at the cashier automatically. 

They made that printing part work, **but are concerned by the customer being confused or not finding what they want if they have a single top level list with tens and tens of items listed, compared to a neatly sorted POS with folders that guide the customer to items sorted by type** or whatever other category they want.

Seems like a reasonable request, and one that makes sense.

**Comment by bolatovumar** [comment]
Sounds like what's needed here is the ability to group items by categories (i.e., you should be able to "tag" an item with "burger" or "drink" or whatever) which then can be used to adjust how these items are displayed.

Additionally, if this feature exists and once https://github.com/btcpayserver/btcpayserver/pull/4102 is merged, users will be able to fetch store items through Greenfield API and build their own frontend to display their store items in however manner they please.

**Reply by pavlenex** [reply]
@bolatovumar Tiankii, is a heavy modified fork of BTCPay, they introduced product categories like this, unsure if it helps https://youtu.be/hDhQvAyEZ-8?t=497

**Comment by EscapeToElSalvador** [comment]
So to take this a step further, popular POS solutions like Micros and Aloha (and many others) use categories, like what you describe - and a concept of 'modifiers'. For an easy example consider a pizza shop that has an item, "3 topping large pizza", and it has a set price, say $15. When you add it to your cart, a picker (if you've enabled modifiers) would pop up with all the available toppings, and the customer or service staff can select from the list the appropriate modifications. Some might even change the price (add chicken for +$1). Then when in the receipt view, you'll see the primary item and below it (tabbed perhaps for readability) you'll see the modifiers.

3 topping large pizza ($15)
   peppers
   onions
   chicken (+1)
Subtotal ($16)
